encoding/Data/ASN1/Prim.hs

```diff
@@ -219,6 +228,49 @@ getIntegerRaw typestr s
             v1 = s `B.index` 0
             v2 = s `B.index` 1
 
+getDouble :: ByteString -> Either ASN1Error ASN1
+{-# INLINE getDouble #-}
+getDouble s = Real <$> getDoubleRaw (B.length s) s
```

Comment:
I haven't seen any other use of getDoubleRaw, so that seems a bit dangerous/unnecessary to expose the length of `s` as a parameter instead of just `where len = B.length s`.

Reply:
Good point, I've pushed the length computation into getDoubleRaw.

Reply:
thanks ! note that now the length is "computed" twice (once in null and once in length); it's fine here since it's o(1) anyway and unlikely to be a fast path :)